src/WebJobs.Script/Host/HostFunctionMetadataProvider.cs

```diff
@@ -37,8 +38,10 @@ public HostFunctionMetadataProvider(IOptionsMonitor<ScriptApplicationHostOptions
         public ImmutableDictionary<string, ImmutableArray<string>> FunctionErrors
            => _functionErrors.ToImmutableDictionary(kvp => kvp.Key, kvp => kvp.Value.ToImmutableArray());
 
-        public ImmutableArray<FunctionMetadata> GetFunctionMetadata(IEnumerable<RpcWorkerConfig> workerConfigs, bool forceRefresh)
+        public async Task<ImmutableArray<FunctionMetadata>> GetFunctionMetadataAsync(IEnumerable<RpcWorkerConfig> workerConfigs, bool forceRefresh)
```

Comment:
This is async because its interface IFunctionMetadataProvider has async method for another implementation - WorkerFunctionMetadataProvider with await calls.